### PR TITLE
Integrate copyright element

### LINKED ISSUE #456: Image Copyright
**Is your feature request related to a problem? Please describe.**
Ensure the copyright holder of non-BBC images is displayed and read correctly by screen readers.

**Describe the solution you'd like**
- Overlay the copyright using text (as opposed to having ImageChef add the copyright as a watermark).
- The visible copyright holder should have a contrast of 4.5:1

### PR DESCRIPTION
**Pull request by bcmn**
Resolves #456 #655

* Integrates copyright component
* Changes fixture data image, to one more typical -- previous had embedded source & logo. [Previous](https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png) | [New](https://ichef.bbci.co.uk/news/640/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg)
* Changes `test/c0000000002o.json` & `test/c0000000025o.json` to allow for testing of image copyright with & without image caption.
* Tested with Voiceover to ensure AT pauses suitably between the copyright text & a following caption or paragraph, & reads the visually hidden ["image source" text](https://github.com/bbc/simorgh/issues/456#issuecomment-420678019) with a suitable pause.
* Capitalises text in copyright element using CSS, to avoid AT reading it incorrectly
* Transparent background, ensures hidden accessibility full-stop is wholly transparent to avoid showing against the translucent background

- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by greenc05** [comment]
@bcmn lets chat about how we can implement the new accessibility copyright requirements when we know what they are

**Comment by bcmn** [comment]
Tested with Voiceover this morning to make sure the ["image source" wording](https://github.com/bbc/simorgh/issues/456#issuecomment-420678019) has suitable pauses, etc. 👍

**Review by sareh — DISMISSED** [review]
Looks great! Thanks for making the changes.

File: src/app/components/Figure/Copyright/index.jsx
Comment (jtart) [inline src/app/components/Figure/Copyright/index.jsx]: As we are already passing `children` in, it might be cleaner to pass `VisuallyHiddenText` in as a child. This would allow us to continue to export the styled Copyright component, and would mean that we can handle things like translations (of `Image source`) at a container rather than component level.

What do you think?

Comment (bcmn) [reply src/app/components/Figure/Copyright/index.jsx]: This is a good point -- from a consistency viewpoint, it seems stuff like ImageCaption hasn't currently split out VisuallyHiddenText to the container like this? It seems like the most similar example I could find in the codebase. 

There's already a ticket for adding the internationalisation to the children, as it's always English (#623) -- perhaps this would fit in there nicely?

Comment (jtart) [reply src/app/components/Figure/Copyright/index.jsx]: Yep that sounds fine. Let's just do it closer to the time and not let it hold us up now.

File: data/test/news/c0000000002o.json
Comment (sareh) [inline data/test/news/c0000000002o.json]: I'd suggest making this change on another test article, for example `c0000000005o`, since the images on this article have another logo in the bottom right.
Also, could you please update this table in the Readme, explaining that there are images with Image source attribution on the associated articles: https://github.com/bbc/simorgh/blob/latest/data/README.md

File: data/prod/news/c0000000025o.json
Comment (sareh) [inline data/prod/news/c0000000025o.json]: @bcmn The production articles are ones which we will *actually* put in production - they'll be audience-facing. For this reason, we shouldn't change the image attribution to be inaccurate.  

@jimjohnsonrollings Jim - is there an example of an image on the Prod articles that have non-BBC attribution, which we could use for testing?

**Review by sareh — CHANGES_REQUESTED** [review]
The work on the PR looks great. I am just wary of changing the content of a Prod article in a way that makes it no longer accurate. Have cc'd in Jim to give his thoughts on the matter.

File: data/prod/news/c0000000025o.json
Comment (jimjohnsonrollings) [reply data/prod/news/c0000000025o.json]: Yes - 25 has both BBC and non-BBC images in.

Comment (sareh) [reply data/prod/news/c0000000025o.json]: Great, we'll add them, then.

Comment (bcmn) [reply data/prod/news/c0000000025o.json]: Updated prod pages to reflect real copyrightHolders & changed cypress tests to reflect.

File: data/test/news/c0000000002o.json
Comment (bcmn) [reply data/test/news/c0000000002o.json]: Changed to `c0000000005o` & updated readme.

File: data/prod/news/c0000000025o.json
Comment (jtart) [inline data/prod/news/c0000000025o.json]: Are these meant to be in all caps?

**Review by jtart — DISMISSED** [review]

File: data/prod/news/c0000000025o.json
Comment (jtart) [reply data/prod/news/c0000000025o.json]: @jimjohnsonrollings ^

Comment (jimjohnsonrollings) [reply data/prod/news/c0000000025o.json]: They will be mixed case in the data - we should however always display them in upper case.

**Comment by bcmn** [comment]
Fixed test/prod data, Figure container now capitalises the copyright string before passing to Copyright component.

File: src/app/containers/Figure/index.jsx
Comment (sareh) [inline src/app/containers/Figure/index.jsx]: I recommend using the CSS property `text-transform: uppercase;` in the Copyright component, rather than changing the actual text from mixed cases to upper case. This is so that screenreaders don't read the text as initials, but rather the words.

File: data/test/news/c0000000005o.json
Comment (sareh) [inline data/test/news/c0000000005o.json:162]: Thanks for updating a Test article to capture the two different cases - an image with caption and without a caption, but both having the copyright text.  
Great that you've also updated the Readme file as well. 💯

**Review by sareh — COMMENTED** [review]
Great work! You've added test cases to our test data, which is very useful, too. 
I've just got one comment about the implementation of the uppercasing.

**Review by sareh — DISMISSED** [review]
Thanks for the changes. Looks great!

**Review by BogdanDogaru — DISMISSED** [review]
Looks good to me. Nice work!

**Comment by bcmn** [comment]
A few screenshots to document...

Third-party copyright, no caption:
screen shot 2018-09-19 at 10 49 50

Third-party copyright, with caption:
screen shot 2018-09-19 at 10 49 32

First-party copyright, no caption:
screen shot 2018-09-19 at 10 49 59

**Comment by bcmn** [comment]
UX reviewed this with Ngan Thi, noticed that whilst everything else seems correct, the background-colour was intended to have a 75% opacity. Currently the background-colour is entirely opaque.

**Comment by sareh** [comment]
@bcmn Looks great.  👍 
I'll explain what I did when reviewing this change to the transparency. 

I ran the lighthouse tests against this branch `npm run test:lighthouse`. This runs against article http://localhost:7080/news/articles/c0000000027o It had an accessibility score of 100 still, so passes colour contrast.  
screen shot 2018-09-20 at 08 52 45

I noted that this might have been because of the particular images in that article, so I ran the dev build `npm run dev`, visited that page http://localhost:7080/news/articles/c0000000027o and then removed the src for an image, so there was no background. 
screen shot 2018-09-20 at 08 53 46


There is a very useful Color Contrast picker in Chrome Canary that now has in-built colour contrast guidance. (How to use tutorial is here https://remysharp.com/2017/08/17/contrast-ratio-in-devtools) 
It has a color contrast of 6.98, which more than meets our requirement (4.5) (https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-testers)

screen shot showing a curve for where there's contrast

**Review by sareh — DISMISSED** [review]

**Review by jtart — DISMISSED** [review]
Looks great!

**Comment by bcmn** [comment]
Updated screenshots...

Third-party copyright, no caption:
screen shot 2018-09-20 at 11 15 07

Third-party copyright, with caption:
screen shot 2018-09-20 at 11 15 41

First-party copyright, no caption:
screen shot 2018-09-20 at 11 15 51

**Review by pjlee11 — DISMISSED** [review]

**Review by jtart — DISMISSED** [review]

**Comment by bcmn** [comment]
Chatting with @nataliesmart, we've noticed that the invisible full stop to ensure a suitable pause when using AT offsets the text in a way that impacts the intended design. Once noticed, it's very obvious. I think we were curious if there was another way to handle this.

Personally, now we've got the visually hidden text for "image source", I wonder if we even need to continue worrying about the original impetus for placing the full stop in there? As far as I can recall, it originated out of wanting to avoid using VHT when it didn't seem 100% necessary, which is a boat that has now sailed. Could/should I just add the entirety of the copyright holder to the VHT, with punctuation to ensure a suitable pause afterwards?

⛵️

**Comment by pjlee11** [comment]
@bcmn as spoke in person I think we should keep the VHT seperate to the visible copyright element as it is becoming to intricate. Therefore we should have VHT with the prefix text and a full stop to ensure the accessibility standard are met, and then set aria-hidden and focusable=false on the visual element

**Review by pjlee11 — DISMISSED** [review]
Great work, hopefully no more requirements will change before it's merged  🎉

**Review by jtart — DISMISSED** [review]

**Comment by greenc05** [comment]
@bcmn please could we talk about the visually hidden copyright code change? This has some implications for assistive technology.

**Comment by pjlee11** [comment]
@greenc05 @bcmn can we agree now that the work, unless failing test, can be merged as is then we will address any concerns in a follow up PR. I ask this because the PR has been open and bouncing across the board for almost 2 weeks due to a lack of UX/Accessibility communication and we should look not to hold up merging work. Thanks :)

Update: To clarify I mean the communication between UX and accessibility

**Comment by greenc05** [comment]
Sure @pjlee11

**Review by jtart — APPROVED** [review]

**Review by sareh — APPROVED** [review]

**Pull request merged**
